ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -340,60 +347,163 @@ azure_object_size(
+    // lock_guard will unlock automatically when the scope is exited.
+    {
+        std::lock_guard<std::mutex> lock_guard(azure_fs->fh_mutex);
+        azure_fs->azure_fh.push_back(azure_fh);
+    }
+    *file_handlep = &azure_fh->fh;
+
     WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(file_type);
-    WT_UNUSED(flags);
-    WT_UNUSED(file_handlep);
+    WT_UNUSED(size);
 
     return 0;
 }
 
+// File handle close.
 static int
 azure_file_close(WT_FILE_HANDLE *file_handle, WT_SESSION *session)
 {
-    WT_UNUSED(file_handle);
+    azure_file_handle *azure_fh = reinterpret_cast<azure_file_handle *>(file_handle);
+    // If there are other active instances of the file being open, do not close file handle.
+    if (--azure_fh->reference_count != 0) {
+        return 0;
+    }
+
+    // No more active instances of open file, close the file handle.
+    azure_file_system *azure_fs = azure_fh->fs;
+    {
+        std::lock_guard<std::mutex> lock_guard(azure_fs->fh_mutex);
+        // Erase-remove idiom to elimate specific file handle
+        azure_fs->azure_fh.erase(
+          std::remove(azure_fs->azure_fh.begin(), azure_fs->azure_fh.end(), azure_fh),
+          azure_fs->azure_fh.end());
+    }
+
     WT_UNUSED(session);
 
     return 0;
 }
 
+// Lock/unlock a file.
 static int
 azure_file_lock(WT_FILE_HANDLE *file_handle, WT_SESSION *session, bool lock)
 {
+    // Locks are always granted.
     WT_UNUSED(file_handle);
     WT_UNUSED(session);
     WT_UNUSED(lock);
 
     return 0;
 }
 
+// Read a file using Azure connection class read object functionality.
 static int
 azure_file_read(
   WT_FILE_HANDLE *file_handle, WT_SESSION *session, wt_off_t offset, size_t len, void *buf)
 {
-    WT_UNUSED(file_handle);
+    azure_file_handle *azure_fh = reinterpret_cast<azure_file_handle *>(file_handle);
+    azure_file_system *azure_fs = azure_fh->fs;
+
+    int ret;
+    if ((ret = azure_fs->azure_conn->read_object(azure_fh->name, offset, len, buf) != 0)) {
+        std::cerr << "azure_file_read: read_object request to Azure failed." << std::endl;
+        return ret;
+    }
     WT_UNUSED(session);
-    WT_UNUSED(offset);
-    WT_UNUSED(len);
-    WT_UNUSED(buf);
 
     return 0;
 }
 
+// Get the size of a file in bytes, by file handle.
 static int
 azure_file_size(WT_FILE_HANDLE *file_handle, WT_SESSION *session, wt_off_t *sizep)
 {
-    WT_UNUSED(file_handle);
-    WT_UNUSED(session);
-    WT_UNUSED(sizep);
-
-    return 0;
+    azure_file_handle *azure_fh = reinterpret_cast<azure_file_handle *>(file_handle);
+    int ret;
+    bool exists;
+    size_t size = 0;
+    if ((ret = azure_fh->fs->azure_conn->object_exists(azure_fh->name, exists, size)) != 0) {
+        std::cerr << "azure_file_open: object_exists request to Azure failed." << std::endl;
+        return ret;
+    }
+    if (exists == false) {
+        std::cerr << "azure_file_open: no such file." << std::endl;
```

Comment:
Same thing with printing the file name.

Reply:
removed the `exists == false` check because its redundant. A file handle is passed in as a parameter. We wouldn't have been able to create the file handle in the first place if the file didn't exist.